src/libprojectM/ProjectM.cpp

```diff
@@ -656,39 +613,11 @@ void ProjectM::destroyPresetTools()
 {
     m_activePreset.reset();
     m_activePreset2.reset();
-
-    if (m_presetPos)
-    {
-        delete (m_presetPos);
-    }
-
-    m_presetPos = 0;
-
-    if (m_presetChooser)
-    {
-        delete (m_presetChooser);
-    }
-
-    m_presetChooser = 0;
-
-    if (m_presetLoader)
-    {
-        delete (m_presetLoader);
-    }
-
-    m_presetLoader = 0;
-
-    if (m_matcher)
-    {
-        delete m_matcher;
-        m_matcher = NULL;
-    }
-
-    if (m_merger)
-    {
-        delete m_merger;
-        m_merger = NULL;
-    }
+    m_presetPos.reset();
```

Comment:
Makes more sense to just have all resource handling logic in the ctor and dtor, then we don't have to think about resetting the member pointers.

Reply:
Ideally, yes. Most of this stuff should probably be cleaned up in other ways as well, but that's definitely something for later.

Reply:
Looked at it again, there's no need to do that manually, so I've just removed the whole function and let the `unique_ptr` dtor take care of it automatically.